app/HieWrapper.hs

```diff
@@ -85,20 +83,18 @@ run opts = do
 
   let
     hieBin = "hie-" ++ ghcVersion
-    backupHieBin = "hie-" ++ reverse (tail $ dropWhile (/='.') $ reverse ghcVersion)
+    backupHieBin = "hie-" ++ init (dropWhileEnd (/='.') ghcVersion)
+    candidates' = [hieBin, backupHieBin, "hie"]
+    candidates = map (++ exeExtension) candidates'
 
-  logm $ "hie exe candidates :" ++ show (hieBin,backupHieBin)
-
-  let candidates' = [hieBin, backupHieBin, "hie"]
-      candidates = map (++ exeExtension) candidates'
+  logm $ "hie exe candidates :" ++ show candidates
```

Comment:
The previous impl didn't list all 3 candidates that the program is searching for. Also switched to `dropWhileEnd` to avoid the need to reverse twice.